packages/world/ts/config/v2/worldWithShorthands.ts

```diff
@@ -65,12 +68,14 @@ export function resolveWorldWithShorthands<world extends WorldWithShorthandsInpu
   const tables = mapObject(world.tables ?? {}, (table) => {
     return isTableShorthandInput(table) ? resolveTableShorthand(table, scope) : table;
   });
-  const namespaces = mapObject(world.namespaces ?? {}, (namespace) => ({
-    ...namespace,
-    tables: mapObject(namespace.tables ?? {}, (table) => {
-      return isTableShorthandInput(table) ? resolveTableShorthand(table, scope) : table;
-    }),
-  }));
+  const namespaces = world.namespaces
+    ? mapObject(world.namespaces, (namespace) => ({
```

Comment:
can we do this same conditional for `tables` to avoid setting it?

Reply:
Yes, and we need to or else the later `hasOwnKey(world, "table")` check will always pass!